packages/flutter_tools/bin/xcode_backend.sh

```diff
@@ -52,9 +52,9 @@ BuildApp() {
   local build_mode="$(echo "${FLUTTER_BUILD_MODE:-${CONFIGURATION}}" | tr "[:upper:]" "[:lower:]")"
   local artifact_variant="unknown"
   case "$build_mode" in
-    release*) build_mode="release"; artifact_variant="ios-release";;
-    profile*) build_mode="profile"; artifact_variant="ios-profile";;
-    debug*) build_mode="debug"; artifact_variant="ios";;
+    *release*) build_mode="release"; artifact_variant="ios-release";;
```

Comment:
What are the actual possible values for the FLUTTER_BUILD_MODE here?  Seems like it would be better to do something like:

```bash
  foo-release* | bar-release*) ...;;
```

So that we would catch invalid prefixes that still include "release" in the name.

Reply:
FLUTTER_BUILD_MODE isn't always set.  And CONFIGURATION comes through with a user defined prefix if flavors are used, but I'm not 100% sure about how to detect them.  TARGETNAME seems to be it, but that also gets set to Runner and I don't think we should special case that name in case someone uses it for an Add2App target

Reply:
OK, I guess this works.  It just seems a bit broader than it needs to be.

Reply:
@xster  or @cbracken is there some other env var I could be checking for this?  I didn't see one that looked reliable enough.